src/expandProperty.js

```diff
@@ -244,6 +294,189 @@ function parseGap(value) {
+    ) {
+      // invalid value, ignore both values
+      return {}
+    }
+    if (alignContent === 'baseline') {
+      // ..., unless that value is a <baseline-position> in which case it is defaulted to start.
+      justifyContent = 'start'
+    } else {
+      // ..., it is copied from the first value
+      justifyContent = alignContent
+    }
+  }
+
+  return {
+    alignContent,
+    justifyContent
+  }
+}
+
+// https://w3c.github.io/csswg-drafts/css-align/#place-items-property
+function parsePlaceItems(value) {
+  // This shorthand property sets both the align-items and justify-items properties in a single declaration. The first value is assigned to align-items. The second value is assigned to justify-items; if omitted, it is copied from the first value.
+  const [alignItems, justifyItems = alignItems] = splitShorthand(value);
+  return {
+    alignItems,
+    justifyItems
+  }
+}
+
+function parsePlaceSelf(value) {
+  // https://w3c.github.io/csswg-drafts/css-align/#place-self-property
+  const [alignSelf, justifySelf = alignSelf] = splitShorthand(value);
+  return {
+    alignSelf,
+    justifySelf
+  }
+}
+
+// https://w3c.github.io/csswg-drafts/css-transitions/#transition-shorthand-property
+// Since transition can accept multiple values, we will run extraction on each value
+const extractTransition = (value) => {
+  const values = splitShorthand(value)
+
+  if (values.length === 1) {
+    if (GLOBAL_VALUES.includes(values[0])) {
+      return {
+        transitionProperty: values[0],
+        transitionDuration: values[0],
+        transitionDelay: values[0],
+        transitionTimingFunction: values[0]
+      }
+    } else if (values[0] === 'all' || values[0] === 'none') {
+      return {
+        transitionProperty: values[0],
+        transitionDuration: '0s', // https://w3c.github.io/csswg-drafts/css-transitions/#transition-duration-property
+        transitionDelay: '0s', // https://w3c.github.io/csswg-drafts/css-transitions/#transition-delay-property
+        transitionTimingFunction: 'ease' // https://w3c.github.io/csswg-drafts/css-transitions/#transition-timing-function-property
+      }
+    }
+  }
+
+  // Note that order is important within the items in this property: the first value that can be parsed as a time is assigned to the transition-duration, and the second value that can be parsed as a time is assigned to transition-delay.
+  const [
+    transitionProperty,
+    transitionDuration = '0s',
+    transitionDelay = '0s',
+    transitionTimingFunction = 'ease'
+  ] = values
+
+  return {
+    transitionProperty,
+    transitionDuration,
+    transitionDelay,
+    transitionTimingFunction
+  }
+}
+// https://w3c.github.io/csswg-drafts/css-transitions/#transition-shorthand-property
+function parseTransition(value) {
+  if (!value.includes(',')) {
+    // single value, let's do fast path
+    return extractTransition(value)
+  }
+
+  const transitions = value.split(',').map(extractTransition)
```

Comment:
Both `transition: color 1s` and `transition: color 1s, margin 0.5s` are valid, and we need to extract them separately.

But the former case (single value) is way more common. So in most cases, we can skip split and merge directly.

`extractTransition` uses the `splitShorthand` function which will do trim, so I use `,` as the separator directly here.